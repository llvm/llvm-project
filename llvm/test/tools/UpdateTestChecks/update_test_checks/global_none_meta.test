## Basic test checking that update_test_checks.py --check-globals none works correctly
# RUN: cp -f %S/Inputs/global_none_meta.ll %t.ll && %update_test_checks %t.ll --check-globals none
# RUN: diff -u %t.ll %S/Inputs/global_none_meta.ll.expected
