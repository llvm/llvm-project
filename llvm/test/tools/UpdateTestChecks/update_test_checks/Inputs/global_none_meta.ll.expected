; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --check-globals none --version 5
; RUN: opt < %s -S | FileCheck %s

; Function Attrs: convergent noinline nounwind
declare hidden spir_func void @__cxx_global_var_init() #0

; Function Attrs: convergent noinline nounwind
define hidden spir_kernel void @_GLOBAL__sub_I_global_init.clcpp() #0 !kernel_arg_addr_space !4 !kernel_arg_access_qual !4 !kernel_arg_type !4 !kernel_arg_base_type !4 !kernel_arg_type_qual !4 {
; CHECK-LABEL: define hidden spir_kernel void @_GLOBAL__sub_I_global_init.clcpp(
; CHECK-SAME: ) #[[ATTR0:[0-9]+]] {{.*}}{
; CHECK-NEXT:  [[ENTRY:.*:]]
; CHECK-NEXT:    call spir_func void @__cxx_global_var_init()
; CHECK-NEXT:    ret void
;
entry:
  call spir_func void @__cxx_global_var_init()
  ret void
}

attributes #0 = { convergent noinline nounwind "no-trapping-math"="true" "stack-protector-buffer-size"="8" }

!llvm.module.flags = !{!0}
!opencl.ocl.version = !{!1}
!opencl.cxx.version = !{!2}
!opencl.spir.version = !{!1}
!llvm.ident = !{!3}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 2, i32 0}
!2 = !{i32 1, i32 0}
!3 = !{!"clang version 20.0.0git"}
!4 = !{}

